lib/trip_dispatcher.rb

```diff
@@ -29,44 +34,110 @@ def load_users(filename)
       return users
     end
 
-
     def load_trips(filename)
       trips = []
       trip_data = CSV.open(filename, 'r', headers: true,
                                           header_converters: :symbol)
 
       trip_data.each do |raw_trip|
         passenger = find_passenger(raw_trip[:passenger_id].to_i)
+        driver = find_driver(raw_trip[:driver_id].to_i)
 
         parsed_trip = {
           id: raw_trip[:id].to_i,
           passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
+          start_time: Time.parse(raw_trip[:start_time]),
+          end_time: Time.parse(raw_trip[:end_time]),
           cost: raw_trip[:cost].to_f,
-          rating: raw_trip[:rating].to_i
+          rating: raw_trip[:rating].to_i,
+          driver: driver
+
         }
 
         trip = Trip.new(parsed_trip)
         passenger.add_trip(trip)
+        driver.add_trip(trip)
+
         trips << trip
       end
 
       return trips
     end
 
+    def load_drivers(filename)
+      drivers = []
+
+      driver_data = CSV.open(filename, 'r', headers: true, header_converters: :symbol)
+
+      driver_data.each do |driver|
+
+        input = {}
+        input[:id] = driver[0].to_i
+        input[:vin] = driver[1]
+        input[:status] = driver[2].to_sym
+        input[:name] = find_passenger(input[:id]).name
+        input[:phone_number] = find_passenger(input[:id]).phone_number
+
+        drivers << Driver.new(input)
+      end
+
+      return drivers
+    end
+
     def find_passenger(id)
       check_id(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
 
+
+    def find_driver(id)
+      check_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
+
     def inspect
       return "#<#{self.class.name}:0x#{self.object_id.to_s(16)} \
               #{trips.count} trips, \
               #{drivers.count} drivers, \
               #{passengers.count} passengers>"
     end
 
+    def request_trip(user_id)
+
+      passenger = find_passenger(user_id)
+
+      this_driver = @drivers.find do |driver|
+        if driver.id != user_id
+          driver.status == :AVAILABLE
+        end
```

Comment:
I'm also not 100% sure this code will do the right thing if `driver.id == user_id`. I think it would be more clear to say:

```ruby
this_driver = @drivers.find do |driver|
  driver.id != user_id && driver.status == :AVAILABLE
end
```